R/utils.R

```diff
@@ -328,6 +328,192 @@ date_to_numeric_modulus <- function(dt, date_column, timestep) {
+#' enw_unset_cache()
+#'
+#' enw_unset_cache(enw_set_cache(file.path(tempdir(), "test")))
+enw_unset_cache <- function() {
+  prior_location <- Sys.getenv("enw_cache_location")
+  if (prior_location != "") {
+    cli::cli_inform(c(
+      i = paste0(
+        "Removing `enw_cache_location = {prior_location}` from .Renviron and ",
+        "from the local environment."
+      )
+    ))
+    Sys.unsetenv("enw_cache_location")
+
+    clean_environ <- enw_get_environment_contents(
+      remove_enw_cache_location = TRUE
+    )
+
+    writeLines(clean_environ$env_contents, clean_environ$env_path)
+
+    invisible(enw_get_environment_contents(remove_enw_cache_location = TRUE))
+  } else {
+    cli::cli_inform(c(
+      "!" = paste0(
+        "`enw_cache_location` not set. ",
+        "Nothing to remove from .Renviron or the local environment."
+      )
+    ))
+  }
+
+  return(invisible(prior_location))
+}
+
+#' Retrieve Stan cache location
+#'
+#' Retrieves the user set cache location for Stan models. This
+#' path can be set through the `enw_cache_location` function call.
+#' If no environmental variable is available the output from
+#' `tempdir` will be returned.
+#'
+#' @return a string representing the file path for the cache location
+#' @importFrom cli cli_inform
+#' @family utils
+#' @export
+enw_get_cache <- function() {
+  cache_location <- Sys.getenv("enw_cache_location")
+
+  if (check_environment_setting(cache_location)) {
+    cache_location <- tempdir()
+    cli::cli_inform(c(
+      "!" = "`enw_cache_location` not specified. Using `tempdir` at {cache_location}" # nolint line_length
+    ))
+  } else {
+    cli::cli_inform(c(i = "Using `{cache_location}` for the cache location."))
+  }
+
+  return(cache_location)
+}
+
+#' Check environment setting
+#'
+#' This internal function checks whether a given environment variable is set or
+#' not. It returns `TRUE` if the variable is either null or an empty string,
+#' indicating that the environment variable is not set. Otherwise, it returns
+#' `FALSE`.
+#'
+#' @param x The environment variable to be checked.
+#'
+#' @return Logical value indicating whether the environment variable is not set
+#' (either null or an empty string).
+#' @keywords internal
+check_environment_setting <- function(x) {
+  return(is.null(x) || x == "")
+}
+
+#' Identify cache location
+#'
+#' This function retrieves environment variable settings and manages the
+#' `.Renviron` file in the user's home directory.
+#' It can optionally remove the entry for `enw_cache_location`.
+#'
+#' @param remove_enw_cache_location Logical indicating whether to remove the
+#' `enw_cache_location` entry from the `.Renviron` file. Defaults to `TRUE`.
+#'
+#' @return A list containing the contents of the `.Renviron` file and its path.
+#' @keywords internal
+enw_get_environment_contents <- function(remove_enw_cache_location = TRUE) {
+  env_location <- Sys.getenv("HOME")
+  env_path <- file.path(env_location, ".Renviron")
```

Comment:
What happens if people are using a local/project-specific .Renviron?

Reply:
Need to fix this. I'll look locally first, then go to `$HOME`

Reply:
https://github.com/epinowcast/epinowcast/pull/416/commits/aa41b0fd120e178012cab9bad5f3e98e0e5f75a2 should fix this piece and testing locally with an empty project with an .Renviron